File: src/test/java/com/cisco/td/general/cocroach/DateIntervalTest.java
package com.cisco.td.general.cocroach;

import org.joda.time.LocalDate;
import org.junit.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

public class DateIntervalTest {
    @Test
    public void containsCorrectDates() throws Exception {
        DateInterval interval = DateInterval.year(2021);

        assertThat(interval.contains(new LocalDate(2020, 12, 31)), is(false));
        assertThat(interval.contains(new LocalDate(2021, 1, 1)), is(true));
        assertThat(interval.contains(new LocalDate(2021, 6, 14)), is(true));
        assertThat(interval.contains(new LocalDate(2021, 12, 31)), is(true));
        assertThat(interval.contains(new LocalDate(2022, 1, 1)), is(false));
    }
}

File: src/test/kotlin/cz/solutions/cockroach/DateIntervalTest.kt
package cz.solutions.cockroach

import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.joda.time.LocalDate
import org.junit.Test

internal class DateIntervalTest {
    @Test
    fun `contains correct dates`() {
        val interval = DateInterval.year(2021)
        assertThat(interval.contains(LocalDate(2020, 12, 31)), `is`(false))
        assertThat(interval.contains(LocalDate(2021, 1, 1)), `is`(true))
        assertThat(interval.contains(LocalDate(2021, 6, 14)), `is`(true))
        assertThat(interval.contains(LocalDate(2021, 12, 31)), `is`(true))
        assertThat(interval.contains(LocalDate(2022, 1, 1)), `is`(false))
    }
}